### PR TITLE
Create script for issue submission (canonical data version changes)

### LINKED ISSUE #1670: Create script for automated issue submission for new exercises and test updates
I suggest to create a script that opens up new issues for needed test updates or new exercises entered into the [problem specification repo](https://github.com/exercism/problem-specifications).

GitHub offers [query parameters](https://help.github.com/en/articles/about-automation-for-issues-and-pull-requests-with-query-parameters) to open up new issues and fill it with title, body, labels...
Existing issues could be checked via the [GitHub API](https://api.github.com/) to avoid duplicates.

I would gladly work on that if you would welcome such a tool for maintaining this track.
What do you think @exercism/java?

### PR DESCRIPTION
**Pull request by lemoncurry**
Closes #1670

Suggested script using GitHub API to create new issue when difference between track exercise version file number and canonical data version number is detected.

I based the script on the [canonical data check script](scripts/canonical_data_check.sh).

I introduced a new label to filter for duplicates, `exercise version update`.

The script adds a generic title, body and labels.
For example:

image

Please have a look at the generic texts and fell free to add or remove information/wording (I'm not a native speaker of English, so I would be very glad for corrections :smile:). I tried to incorporate parts of texts that have been previously used.

The script may be tested with a private repository. For authentication you need to create a [token](https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line), which should be kept in a hidden file in your `home directory`, and of course should never be committed. Further explanation on what to put into the file is described in the script.

Right now, the script only checks for changes in the test version, but not for new exercises.
I'd suggest to add this additional functionality at a later point.

Please let me know if you have further improvements or ideas for the script @exercism/java  :smile:

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

File: scripts/create_issues_versionchange_canonical.sh [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+track_exercise_slugs=$(jq '.exercises[] | select(has("deprecated") | not) | .slug' "$config_file_path" | tr -d "\"" | sort)
+update_needed_count=0
+
+
+# Create a file named ".exercism-version-update-issue-script-settings.sh"
+# with the following variables and put in in your home directory
+#
+# TOKEN="your_token"
+# OWNER="your_username"
+# REPO="repo_name"
```
Comment (FridaTveit) [inline scripts/create_issues_versionchange_canonical.sh]: Printing an error with a summary of this information if these values aren't present might be useful? 🙂

File: scripts/create_issues_versionchange_canonical.sh [hunk 2]
```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+
+print_usage() {
+    echo ">>>>> Usage: ./canonical_data_check_open_issue.sh -t path/to/track -s path/to/problem/specifications"
```
[the remark this answers is not in the document]
Comment (lemoncurry) [reply scripts/create_issues_versionchange_canonical.sh]: Thank you @FridaTveit , I obviously forgot to change that :disappointed:

File: scripts/create_issues_versionchange_canonical.sh (hunk 1 above)
Comment (lemoncurry) [reply scripts/create_issues_versionchange_canonical.sh]: Great idea! I will add the error message.

**Comment by lemoncurry** [comment]
I will add some documentation for the script to the contribution guide in the following days.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -192,3 +193,32 @@ To run this script:
...
+## Checking tests are up to date and submit new issues
+
+There is [a script which allows you to submit new issues](https://github.com/exercism/java/blob/master/scripts/create_issues_versionchange_canonical.sh) to this repo with generic title, description and labels if a change in version was detected.
+
+Example generic new isse:
+<img width="1005" alt="image" src="https://user-images.githubusercontent.com/6614867/57221803-bf1a6600-7000-11e9-93cf-b930ef24ce97.png">
+
+Before you may submit a new issue, the script
+  1. Checks for differences between version numbers of each exercise (in comparison with the version number of the canonical data)
+  2. Checks whether an open issue for this exercise; if there is an open issue, you will have to check by yourself if the title of the open issue might be changed to include the new version number. Here, it is important to check whether someone is already working on the issue.
```
Comment (FridaTveit) [inline CONTRIBUTING.md]: Maybe "Checks whether an open issue exists for this exercise"? 🙂

File: scripts/create_issues_versionchange_canonical.sh [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        update_needed_count=$((update_needed_count + 1))
+        read -p "Press enter to continue."
+        echo
+        continue
+    fi
+
+    track_data_version=$(cat "$track_exercise_version_file_path" | sed 's/\r$//')
+
+    if [[ "$track_data_version" = "$canonical_data_version" ]]; then
+        # echo "$slug: up-to-date."
```
Comment (FridaTveit) [inline scripts/create_issues_versionchange_canonical.sh]: As this is commented out, should this line be removed? 🙂

**Comment by lemoncurry** [comment]
Thank you @FridaTveit for your review

**Pull request merged**